cpp/src/Ice/TimeUtil.cpp

```diff
@@ -15,19 +15,23 @@ IceInternal::timePointToString(const chrono::system_clock::time_point& timePoint
     time_t time = static_cast<time_t>(chrono::duration_cast<chrono::seconds>(timePoint.time_since_epoch()).count());
     struct tm tr;
 #ifdef _MSC_VER
-    auto p = localtime_s(&tr, &time);
-#else
-    auto p = localtime_r(&time, &tr);
-#endif
+    // localtime_s in Microsoft CRT is incompatible with the C standard as it returns errno_t instead of struct tm*
+    errno_t p = localtime_s(&tr, &time);
```

Comment:
We introduced this issue when fixing the lint for ignoring the return values of this functions. It is much clear that both return different types when not using `auto`.